Sources/BoostBLEKit/HubProperty.swift

```diff
@@ -8,56 +8,120 @@
+        case uint8(UInt8)
+        case int8(Int8)
         case string(String)
-        case integer(Int)
-        
-        public var stringValue: String {
-            switch self {
-            case .string(let string):
-                return string
-            case .integer(let integer):
-                return "\(integer)"
-            }
-        }
-        
-        public var intValue: Int? {
-            switch self {
-            case .string:
-                return nil
-            case .integer(let integer):
-                return integer
-            }
-        }
     }
     
-    public func value(from data: Data) -> Value? {
+    public var encoding: Encoding {
         switch self {
         case .advertisingName:
+            return .string
+        case .button:
+            return .bool
+        case .firmwareVersion:
+            return .versionNumber
+        case .hardwareVersion:
+            return .versionNumber
+        case .rssi:
+            return .int8
+        case .batteryVoltage:
+            return .uint8
+        case .batteryType:
+            return .uint8
+        case .manufacturerName:
+            return .string
+        case .radioFirmwareVersion:
+            return .string
+        case .legoWirelessProtocolVersion:
+            return .lwpVersionNumber
+        case .systemTypeId:
+            return .uint8
+        case .HWNetworkId:
+            return .uint8
+        case .primaryMacAddress:
+            return .macAddress
+        case .secondaryMacAddress:
+            return .macAddress
+        case .hardwareNetworkFamily:
+            return .uint8
+        }
+    }
+    
+    public func value(from data: Data) -> Value? {
+        switch encoding {
+        case .bool:
+            return .bool(data[0] == 0 ? false : true)
+            
+        case .uint8:
+            return .uint8(data[0])
+            
+        case .int8:
+            return .int8(data.withUnsafeBytes({ $0.load(as: Int8.self) }))
+            
+        case .string:
             guard let string = String(data: data, encoding: .utf8) else { return nil }
             return .string(string)
             
-        case .firmwareVersion:
-            guard data.count == 4 else { return nil }
-            var version = data.reversed().map { String(format: "%02x", $0) }.joined(separator: "")
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 1))
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 3))
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 6))
-            return .string(version)
+        case .versionNumber:
+            precondition(data.count == 4)
+            let major = data[3] & 0xF0 >> 4
+            let minor = data[3] & 0x0F
+            let bugfix = data[2] & 0x0F
+            let buildNumber = UInt16(data[1]) << 8 + UInt16(data[0])
+
+            return .string("\(major).\(minor).\(bugfix).\(buildNumber)")
```

Comment:
According to the document, all version numbers are encoded as BCD. For instance, `[0x23, 0x02, 0x00, 0x10]` should be decoded to `1.0.00.0223`.
https://lego.github.io/lego-ble-wireless-protocol-docs/index.html#lwp-ver-no

Could you change your logic?